sympy/codegen/fnodes.py

```diff
@@ -320,7 +320,7 @@ def dimension(*args):
 
 assumed_size = dimension('*')
 
-def array(symbol, dim, intent=None, **kwargs):
+def array(symbol, dim, intent=None, *, attrs=[], value=None, type=None):
```

Comment:
Don't use mutable keyword defaults. `attrs=()` should work here. 

We might even consider if enabling a "no keyword defaults" checker for the tests is a good idea (e.g., using https://pypi.org/project/flake8-mutable/)

Reply:
I'll make this change because its harmless, but as a general rule I think avoiding mutable defaults is side-stepping the real problem. The thing that's dangerous is mutating arguments at all, not having defaults that could be mutated. For instance, if this function _did_ mutate this `attrs` default, then it would also mutate:
```python
attrs = []
array(..., attrs=attr)
```
which would be almost as bad.